### PR TITLE
add a simple AxesGrid example

### PR DESCRIPTION
**Pull request by dennissergeev**
There was a [question](https://groups.google.com/forum/#!topic/scitools-iris/88ahp7upnF8) about using AxesGrid toolkit with cartopy, and I thought it would be useful to have in the example gallery. I hope my approach is correct.

axes_grid_cartopy

### CODE REVIEW

File: lib/cartopy/examples/axes_grid_basic.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    wave = 0.75 * (np.sin(2 * lats) ** 8) * np.cos(4 * lons)
+    mean = 0.5 * np.cos(2 * lats) * ((np.sin(2 * lats)) ** 2 + 2)
+
+    lats = np.rad2deg(lats)
+    lons = np.rad2deg(lons)
+    data = wave + mean
+
+    times = np.linspace(-1, 1, ntimes)
+    data = np.rollaxis(data.repeat(ntimes).
+                       reshape((*data.shape, ntimes)), -1)
```
Comment (QuLogic) [inline lib/cartopy/examples/axes_grid_basic.py]: This is invalid syntax on Python 2.

File: lib/cartopy/examples/axes_grid_basic.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        ax.set_xticks(np.linspace(-180, 180, 5), crs=projection)
+        ax.set_yticks(np.linspace(-90, 90, 5), crs=projection)
+        lon_formatter = LongitudeFormatter(zero_direction_label=True)
+        lat_formatter = LatitudeFormatter()
+        ax.xaxis.set_major_formatter(lon_formatter)
+        ax.yaxis.set_major_formatter(lat_formatter)
+
+        p = ax.contourf(lons, lats, data[i, ...],
+                        transform=projection,
+                        cmap='YlGnBu_r')
```
Comment (QuLogic) [inline lib/cartopy/examples/axes_grid_basic.py]: Diverging data should probably use `RdBu`.

File: lib/cartopy/examples/axes_grid_basic.py [hunk 3]
```diff
@@ -32,8 +32,8 @@ def sample_data_3d(shape):
     data = wave + mean
 
     times = np.linspace(-1, 1, ntimes)
-    data = np.rollaxis(data.repeat(ntimes).
-                       reshape((*data.shape, ntimes)), -1)
+    new_shape = [i for i in data.shape] + [ntimes]
```
Comment (QuLogic) [inline lib/cartopy/examples/axes_grid_basic.py]: Tuples can be added directly: `data.shape + (ntimes, )`

**Comment by QuLogic** [comment]
I think CI failures are all external and can be ignored in this case.

**Comment by dennissergeev** [comment]
By the way, I don't know why I didn't think about it earlier, but is it worth implementing a new cartopy class that inherits `AxesGrid` and builds upon it? E.g.

```python
class GeoAxesGrid(AxesGrid):
    def __init__(self, fig, rect, nrows_ncols, projection, **axesgrid_kw):
        axes_class = (GeoAxes, dict(map_projection=projection))
        axesgrid_kw['label_mode'] = ''
        super(GeoAxesGrid, self).__init__(fig, rect, nrows_ncols,
                                          axes_class=axes_class, **axesgrid_kw)
```

**Comment by marqh** [comment]
Hi @dennissergeev 

firstly, apologies for the delay in getting this adopted, it's a neat example.  There's been some testing headaches on the cartopy master which have held this and numerous other things back.

There is one more step to getting this in, it is a one shot deal, but it is required.
Cartopy requires a Contributor License Agreement, see
http://scitools.org.uk/governance.html

please may you have a look at this and send us a signed copy if you're happy, then I will be most happy to merge this PR

many thanks
mark

**Comment by marqh** [comment]
> By the way, I don't know why I didn't think about it earlier, but is it worth implementing a new cartopy class that inherits AxesGrid and builds upon it? E.g.

this is an interesting topic, and I would like to think about it.  I'm not immediately sure, but I have no strong reservations

may I suggest that you raise a separate issue, referencing this PR, so the idea is openly preserved once this is closed?

thank you
mark

**Comment by dennissergeev** [comment]
Hi @marqh

Thanks for your positive feedback! I've just sent you the CLA.
I will raise a separate issue about the second idea.

Cheers,
Denis

**Comment by marqh** [comment]
@dennissergeev 
many thanks for the contribution

**Pull request merged**